File: building_kotlin_apps/src/main/java/chapter_02/java/Casts.java
package chapter_02.java;

import java.util.List;

public class Casts {

    private int getSize(Object obj) {
        if (obj instanceof String) {
            return ((String) obj).length();
        } else if (obj instanceof Integer) {
            return ((Integer) obj).intValue();
        } else if (obj instanceof List<?>) {
            return ((List) obj).size();
        }
        return 0;
    }
}

File: building_kotlin_apps/src/main/java/chapter_02/kotlin/Casts.kt
package chapter_02.kotlin

fun main() {
    println(getSize(12))
    println(getSize(listOf("AA", "BB")))
    println("Hello Kotlin")
}

fun getSize(obj: Any): Int = when(obj) {
    is String -> obj.length
    is Int -> obj
    is List<*> -> obj.size
    else -> 0
}